utils/codec/codec.go

```diff
@@ -129,19 +110,15 @@ func (c *codec) SetMaxSliceLen(size int) {
 // 8) nil slices are marshaled as empty slices
 
 // To marshal an interface, [value] must be a pointer to the interface
-func (c *codec) Marshal(value interface{}) ([]byte, error) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
+func (c *codec) MarshalInto(value interface{}, p *wrappers.Packer) error {
```

Comment:
We should change this comment above:
```
// 3) To include a field of a struct in the serialized form, add the tag `serialize:"true"` to it
```
to reflect that the tag name/value are now customizable